pkg/controller/utils/utils.go

```diff
@@ -107,12 +112,41 @@ func CheckCloudCredCreation(awsClient ccaws.Client, logger log.FieldLogger) (boo
 	return CheckPermissionsAgainstActions(awsClient, credMintingActions, logger)
 }
 
+// getClientDetails will return the *iam.User associated with the passed in credentials,
+// a boolean indicating whether the user is the 'root' account, and any error encountered
+// while trying to gather the info
```

Comment:
nit: [godoc comments should have trailing punctuation][1].

[1]: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Reply:
+1